app/controllers/urls_controller.rb

```diff
@@ -0,0 +1,85 @@
+class UrlsController < ApplicationController
+  before_action :logged_in_user
+  before_action :correct_user, only: [:edit, :update]
+
+
+  def new
+    @url = Url.new
+  end
+
+  def create
+    @url = Url.new(url_params)
+    @url.user = current_user
+    if @url.save
+      flash[:success] = "Shortened!"
+      render 'land'
+    else
+      flash[:danger] = @url.errors.full_messages.first
+      render 'new'
+    end
+  end
+
+  def land
+    @url = Url.find(params[:id])
+  end
+
+  def index
+    @urls = Url.paginate(page: params[:page])
+    @pageTitle = "List of Links"
+  end
+
+  def myIndex
+    @urls = Url.where(user: current_user).paginate(page: params[:page])
+    @pageTitle = "My Links"
+    render 'index'
+  end
+
+  def show
+    @url = Url.find(params[:id])
+    @url.clicks += 1
+    @url.save
```

Comment:
When this gets super popular, you'll start losing some click increments due to a race condition here. Check out [increment_counter](http://api.rubyonrails.org/classes/ActiveRecord/CounterCache/ClassMethods.html#method-i-increment_counter).